src/bandersnatch_storage_plugins/s3.py

```diff
@@ -0,0 +1,368 @@
+import configparser
+import contextlib
+import datetime
+import hashlib
+import logging
+import os
+import pathlib
+from pathlib import PurePath
+import tempfile
+from typing import (
+    IO,
+    Any,
+    Dict,
+    Generator,
+    List,
+    Optional,
+    Type,
+    Union,
+    TextIO
+)
+
+import boto3
+from botocore.client import Config
+import filelock
+from bandersnatch.storage import PATH_TYPES, StoragePlugin
+from s3path import PureS3Path, S3Path as _S3Path, register_configuration_parameter
+
+logger = logging.getLogger("bandersnatch")
+
+
+class S3Path(_S3Path):
```

Comment:
type annotations ...

Reply:
hmm not sure what you mean, this class doesnt need type annotation, and I just added type annotation for mkdr method if you meant that.